gems/aws-sdk-docdb/lib/aws-sdk-docdb/plugins/cross_region_copying.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'aws-sigv4'
+
+module Aws
+  module DocDB
+    module Plugins
+      # This plugin populates the `:pre_signed_url` request param for APIs
+      # that support cross region copying.
+      #
+      # This parameter is required by RDS when copying an encrypted snapshot
+      # across regions. This plugin will be skipped if the `:pre_signed_url`
+      # parameter is provided by the user.
+      class CrossRegionCopying < Seahorse::Client::Plugin
+        # @api private
+        class Handler < Seahorse::Client::Handler
+          def call(context)
+            params = context.params
+            if params.is_a?(Hash) &&
+               params[:source_region] && !params[:pre_signed_url]
+              params[:pre_signed_url] = presigned_url(context, params)
+            end
+            @handler.call(context)
+          end
+
+          private
+
+          def presigned_url(context, params)
+            # :source_region is not modeled in the api
+            source_region = params.delete(:source_region)
+            param_list = Aws::Query::ParamList.new
+            param_list.set('Action', context.operation.name)
+            param_list.set('DestinationRegion', context.config.region)
+            param_list.set('Version', context.config.api.version)
+            Aws::Query::ParamBuilder.new(param_list).apply(
+              context.operation.input,
+              params
+            )
+            signer = Aws::Sigv4::Signer.new(
+              service: 'rds',
+              region: source_region,
+              credentials_provider: context.config.credentials
+            )
+            url = Aws::Partitions::EndpointProvider.resolve(
+              signer.region, 'rds'
+            )
+            url += "?#{param_list}"
+            signer.presign_url(
+              http_method: 'GET',
+              url: url,
+              expires_in: 3600
+            ).to_s
+          end
+        end
+
+        handler(
+          Handler,
+          step: :initialize,
+          operations: [
+            :copy_db_cluster_snapshot
```

Comment:
Any easy way to have this be set by the customization we do in the generator instead of maintained seperatly here?  I know we don't intend to support additional operations... but I am concerned about the ability for these to get out of sync.

Reply:
the customization in the generator is just middleware to insert API reference changes prior to generation. this list is for the handler list and is separate.